### PR TITLE
Add support for alternate lookup forms and token transforms

### PR DESCRIPTION
**Pull request by craigp-atw**
Before:
image

After:
image

### CODE REVIEW

**Comment by craigp-atw** [comment]
There's no rush for this, just thought I'd get it out as a starting point for some things.

File: app/src/lib/parser/alternate_lookups.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import {TOKEN_TYPE, create_token} from "./token";
```
Comment (longrunningprocess) [inline app/src/lib/parser/alternate_lookups.js:1]:
```suggestion
import {TOKEN_TYPE, create_token} from './token'
```

Comment (longrunningprocess) [reply app/src/lib/parser/alternate_lookups.js:1]: take a look at the `format` scripts in the `package.json` to take care of little stuff like this before committing.

File: app/src/lib/parser/alternate_lookups.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * These words/phrases (and some others) are accepted by the Analyzer as alternates for 
+ * certain words in the Ontology.
+ * This could easily be put into a config file to prevent the need to rebuild if adding a new one.
+ * 
+ * @type {AlternateLookup[]}
+ */
+const ALTERNATE_LOOKUPS = [
+    [['in','order','to'], 'in-order-to'],
```
Comment (longrunningprocess) [inline app/src/lib/parser/alternate_lookups.js:15]: excellent structure!

File: app/src/lib/parser/alternate_lookups.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+     */
+    function create_next_token() {
+        if (ALTERNATE_LOOKUPS_OPENING_TOKENS.has(tokens[current].token.toLowerCase())) {
+            const alternate = ALTERNATE_LOOKUPS.find(matches_alternate)
+            if (alternate) {
+                const [token_list, new_lookup] = alternate
+
+                // Use the real token strings to ensure capitalization stays the same
+                const slice_end = current + token_list.length
+                const combined_tokens = tokens.slice(current, slice_end).map(t => t.token)
```
Comment (longrunningprocess) [inline app/src/lib/parser/alternate_lookups.js:56]:
```suggestion
                const combined_tokens = tokens.slice(current, slice_end).map(({token) => token)
```

Comment (longrunningprocess) [reply app/src/lib/parser/alternate_lookups.js:56]: as an option if you find it a little more concise

File: app/src/lib/parser/token_transforms.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    ['do', ['not'], set_token_type(TOKEN_TYPE.FUNCTION_WORD)],
+    ['because', ['of'], set_token_lookup('because-B')],
+]
+
+/**
+ * @param {TokenType} type 
+ * @returns {(token: Token) => Token}
+ */
+function set_token_type(type) {
+	return (token) => {return {...token, type}}
```
Comment (longrunningprocess) [inline app/src/lib/parser/token_transforms.js:26]:
```suggestion
	return token => ({...token, type})
```

Comment (longrunningprocess) [reply app/src/lib/parser/token_transforms.js:26]: this is another way to write this if you like it.

File: app/src/lib/parser/token_transforms.js
Comment (longrunningprocess) [inline app/src/lib/parser/token_transforms.js:1]: this is sophisticated, very glad to have a linguist involved for stuff like this!!!

File: app/src/lib/parser/token_transforms.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    /**
+     * 
+     * @returns {Token}
+     */
+    function create_next_token() {
+		if (tokens[current].type != TOKEN_TYPE.LOOKUP_WORD) {
+			return tokens[current++]
+		}
+
+        const transform = TOKEN_TRANSFORMS.find(matches_transform)?.[2] ?? ((token) => token)
```
Comment (longrunningprocess) [inline app/src/lib/parser/token_transforms.js:61]:
```suggestion
        const transform = TOKEN_TRANSFORMS.find(matches_transform)?.[2] ?? token => token
```

Comment (longrunningprocess) [reply app/src/lib/parser/token_transforms.js:61]: I'd be curious if my shortened syntax works ok as a right-hand assignment...

**Review by longrunningprocess — CHANGES_REQUESTED** [review]
this is really beginning to level this app up bigtime!

I'd like to see some unit tests for these when you can.

File: app/src/lib/parser/token_transforms.js (hunk 5 above)
Comment (craigp-atw) [reply app/src/lib/parser/token_transforms.js:61]: `(token => token)` works

**Comment by longrunningprocess** [comment]
> this is really beginning to level this app up bigtime!
> 
> I'd like to see some unit tests for these when you can.

after further conversation, it's clear unit test for these won't make sense quite yet.

**Review by longrunningprocess — APPROVED** [review]

**Pull request merged**